nujo/utils/data/dataset.py

```diff
@@ -0,0 +1,42 @@
+from os.path import expanduser
+
+from numpy import array, empty, float32, vstack
+
+from nujo.utils.data.dataset_iterator import DatasetIterator
+
+
+class Dataset:
+    '''
+
+    Parameters:
+    -----------
+    name : will be downloaded from the UCI ML repo
+
+    Returns:
+    --------
+    tuple : stores the csv dataset,
+    - np array with floating point integers
+    - np array with labels
+    '''
+    def __init__(self, name):
+        file = expanduser('~/.nujo/') + name
+        if '.data' not in name:
+            file += '.data'
+        with open(file, 'r+') as data:
+            lines = data.readlines()
+            self.X = empty((0, 4))
+            self.y = empty((0, 1))
+
+        for line in lines[:-1]:
+            x = array([line.split(',')[:4]], dtype=float32)
+            self.X = vstack((self.X, x))
+            y = array([line.split(',')[-1][:-1]], dtype='U')
+            self.y = vstack((self.y, y))
```

Comment:
Move this logic in a separate method. `__init__` methods should only contain variable assignments. Also, save `name` as public attribute.

Make it look something like:
```
def __init__(self, name):
        self.name = name
        ...
        self.X, self.y = self._load_from_file(filename)

def _load_from_file(self, filename):
        ...
```